tutorials/plot_stats.py

```diff
@@ -11,40 +11,126 @@
 # Applying statistical measures to spiking data
 # ---------------------------------------------
 #
-# Words, words, words.
+# Sections
+# ~~~~~~~~
+#
+# This tutorial contains the following sections:
+#
+# 1. Compute and plot different shuffles of spikes
+# 2. Compute empirical p-value from distribution of spikes
+# 3. Compute z-score from distribution of surrogates
 #
 
 ###################################################################################################
 
 # sphinx_gallery_thumbnail_number = 1
 
-import spiketools
+# import auxiliary libraries
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# import all functions from spiketools.stats
+from spiketools.stats.generators import *
+from spiketools.stats.permutations import *
+from spiketools.stats.shuffle import *
 
 ###################################################################################################
 
-# Settings
-...
+# Generate spike times for spikes at 10Hz for 1000 seconds
+poisson_generator = poisson_train(10, 1000)
+# get spike in seconds
+spikes_s = np.array([spike for spike in poisson_generator])
+# convert to milliseconds
+spikes_ms = spikes_s*1000
+spikes_ms = np.unique(spikes_ms.astype(int))
 
 ###################################################################################################
-# Subtitle
+# 1. Compute and plot different shuffles of spikes
 # ~~~~~~~~
 #
-# Words, words, words.
+# Use spikes_ms to compute shuffle_isis (with 10 shuffles), shuffle_bins (with 10 shuffles), 
+# shuffle_poisson (with 10 shuffles), and shuffle_circular (with 10 shuffles).
+# Also compute a vector permutation, for comparison.
 #
 
 ###################################################################################################
 
-# Do some code stuff
-print('Hello World!')
+# Shuffle spike ms using four methods
+shuffled_isis = shuffle_isis(spikes_ms, n_shuffles=10)
+shuffled_bins = shuffle_bins(spikes_ms, bin_width_range=[50, 200], n_shuffles=10)
+shuffled_poisson = shuffle_poisson(spikes_ms, n_shuffles=10)
+shuffled_circular = shuffle_circular(spikes_ms, shuffle_min=200, n_shuffles=10)
+
+# Do a vector permutation, for comparison
```

Comment:
I don't think we want / need to show the usage of `vec_perm` here. This function is basically a sub-function (something we call from with shuffle_poisson) for helping to run permutations, but by itself it's not a full shuffling technique.

Reply:
Related to this, we can see in the plot that it looks a bit funky:
<img width="317" alt="Screen Shot 2021-11-05 at 4 07 07 PM" src="https://user-images.githubusercontent.com/7727566/140572126-03e5cbc6-7352-425d-ac47-c87ff884e290.png">
y: